L1Trigger/Phase2L1ParticleFlow/src/CaloClusterer.cc

```diff
@@ -0,0 +1,640 @@
+      }
+    }
+  }
+  //// consistency check 1: check that find_cell works
+  //for (float teta = 0; teta <= 5.0; teta += 0.02) {
+  //    for (float tphi = -M_PI; tphi <= M_PI; tphi += 0.02) {
+  //        find_cell(+teta, tphi);
+  //        find_cell(-teta, tphi);
+  //    }
+  //}
+}
+
+int l1tpf_calo::Phase1GridBase::find_cell(float eta, float phi) const {
+  int ieta =
+      (eta != 0) ? std::distance(towerEtas_, std::lower_bound(towerEtas_, towerEtas_ + nEta_, std::abs(eta))) : 1;
+  if (ieta == nEta_)
+    return -1;  // outside bounds
+  assert(ieta > 0 && ieta < nEta_);
+  if (ieta > nEta_)
+    ieta = nEta_;
+  if (eta < 0)
+    ieta = -ieta;
+  if (phi > 2 * M_PI)
+    phi -= 2 * M_PI;
+  if (phi < 0)
+    phi += 2 * M_PI;
+  int iphi = std::floor(phi * nPhi_ / (2 * M_PI));
+  if (phi >= 2 * M_PI)
+    iphi = nPhi_ - 1;  // fix corner case due to roundings etc
+  assert(iphi < nPhi_);
+  if (std::abs(ieta) >= ietaVeryCoarse_)
+    iphi -= (iphi % 4);
+  else if (std::abs(ieta) >= ietaCoarse_)
+    iphi -= (iphi % 2);
+  iphi += 1;
+  //if (!valid_ieta_iphi(ieta,iphi)) {
+  //    printf("Error in finding cell for eta %+7.4f phi %+7.4f, got ieta = %+3d iphi %2d which is not valid\n",
+  //        eta, phi, ieta, iphi);
+  //}
+  assert(valid_ieta_iphi(ieta, iphi));
+  int icell = ifind_cell(ieta, iphi);
+  assert(icell != -1);
+
+  //if (std::abs(eta - eta_[icell]) > 0.501*etaWidth_[icell] || std::abs(deltaPhi(phi, phi_[icell])) > 0.501*phiWidth_[icell]) {
+  //    printf("Mismatch in finding cell for eta %+7.4f phi %+7.4f, got ieta = %+3d iphi %2d which has eta %+7.4f +- %.4f phi %+7.4f +- %.4f ; deta = %+7.4f dphi = %+7.4f\n",
+  //        eta, phi, ieta, iphi, eta_[icell], etaWidth_[icell], phi_[icell], phiWidth_[icell], eta - eta_[icell], deltaPhi(phi, phi_[icell]));
+  //}
+  //assert(std::abs(eta - eta_[icell]) <= 0.5*etaWidth_[icell]);
+  //assert(std::abs(deltaPhi(phi, phi_[icell])) <= 0.5*phiWidth_[icell]);
+  return icell;
+}
+
+int l1tpf_calo::Phase1GridBase::imove(int ieta, int iphi, int deta, int dphi) {
+  int ie = ieta, iph = iphi;
+  switch (deta) {
+    case -1:
+      ie = (ie == -nEta_ ? 0 : (ie == +1 ? -1 : ie - 1));
+      break;
+    case +1:
+      ie = (ie == +nEta_ ? 0 : (ie == -1 ? +1 : ie + 1));
+      break;
+    case 0:
+      break;
+    default:
+      assert(false);
+  };
+  if (ie == 0)
+    return -1;
+  switch (dphi) {
+    case -1:
+      iph = (iph == 1 ? nPhi_ : iph - 1);
+      break;
+    case +1:
+      iph = (iph == nPhi_ ? 1 : iph + 1);
+      break;
+    case 0:
+      break;
+    default:
+      assert(false);
+  };
+  if (!valid_ieta_iphi(ie, iph))
+    return -1;
+  int icell = ifind_cell(ie, iph);
+  assert(!(ie == ieta && iph == iphi));
+  assert(icell != -1);
+  assert(icell != ifind_cell(ieta, iphi));
+  return icell;
+}
+
+const l1tpf_calo::Grid *l1tpf_calo::getGrid(const std::string &type) {
+  static Phase1Grid _phase1Grid;
```

Comment:
make these const just to be safe

Reply:
done (though they're always returned as `const *` so it was safe already)